File: jira/frontend-reactjs/src/types/activity.ts
// ============================================================================
// Activity Log Types
// ============================================================================

// ── Enums / Union types ─────────────────────────────────────────────────────

export type ActivityAction =
  | "created"
  | "updated"
  | "deleted"
  | "commented"
  | "status_changed"
  | "assigned"
  | "moved";

export type ActivityEntityType = "project" | "task" | "sprint" | "comment";

// ── Entity ──────────────────────────────────────────────────────────────────

export interface ActivityLog {
  id: string;
  user_id: string;
  user_name: string;
  user_avatar: string | null;
  action: ActivityAction;
  entity_type: ActivityEntityType;
  entity_id: string;
  entity_title: string;
  project_id: string | null;
  project_name: string | null;
  details: Record<string, unknown> | null;
  created_at: string;
}

// ── Backwards-compat alias (the original monolith exported "Activity") ──────

/** @deprecated Use `ActivityLog` instead */
export type Activity = ActivityLog;

// ── Query helpers ───────────────────────────────────────────────────────────

export interface ActivityQueryParams {
  page?: number;
  page_size?: number;
  entity_type?: ActivityEntityType;
  entity_id?: string;
  user_id?: string;
  action?: ActivityAction;
}

File: project-hub/backend/app/schemas/activity.py
from pydantic import BaseModel
from typing import Any


class ActivityResponse(BaseModel):
    id: str
    user_id: str
    user_name: str
    user_avatar: str | None = None
    action: str
    entity_type: str
    entity_id: str
    entity_title: str
    project_id: str | None = None
    project_name: str | None = None
    details: dict[str, Any] | None = None
    created_at: str

    model_config = {"from_attributes": True}
